src/authorize.js

```diff
@@ -37,16 +37,42 @@
       url += '&state=' + encodeURIComponent(redirectUri.substring(hashPos+1));
     }
     url += '&response_type=token';
+
+    if (global.cordova) {
+      return RemoteStorage.Authorize.openWindow(
+          url,
+          redirectUri,
+          'location=yes,clearsessioncache=yes,clearcache=yes'
+        )
+        .then(function(authResult) {
+          remoteStorage.remote.configure({
+            token: authResult.access_token
+          });
+
+          // TODO
+          // sync doesn't start until after reload
+          // possibly missing some initialization step?
+          global.location.reload();
+        })
+        .then(null, function(error) {
```

Comment:
@untitaker `.catch` is what I'd normally use, but I only remember seeing `.then(null,` used elsewhere in the project so I thought that might have been a deliberate style choice and followed it.

@galfert I believe adding a handler to the previous `.then` will cause it to miss any errors triggered in the success callback of the same `.then`. Adding a dedicated error handler at the end of the chain is safer in that sense.